packages-web/accessibility-helper/src/AccessibilityHelper.tsx

```diff
@@ -0,0 +1,124 @@
+                                attributeConditionsRef.current?.splice(index, 1);
+                                target!.removeAttribute(attrEntry.attribute);
+                            }
+                        }
+                    });
+                }
+            } catch (error) {
+                console.error(`Set Attribute Target sector ${props.targetSelector} is not valid`, error);
+            }
+        }
+    }, [
+        props.targetSelector,
+        props.attributesList,
+        attributeValuesRef.current,
+        attributeConditionsRef.current,
+        contentRef.current
+    ]);
+
+    useEffect(() => {
+        if (!isValid(props.targetSelector)) {
+            return;
+        }
+        const isAnythingChanged = props.attributesList.every(attrEntry => {
+            const valueToBeSet =
+                attrEntry.valueSourceType === "expression" ? attrEntry.valueExpression : attrEntry.valueText;
+            return (
+                (valueToBeSet && !attributeValuesRef.current?.some(value => value === valueToBeSet.value)) ||
+                (attrEntry.attributeCondition.value &&
+                    !attributeConditionsRef.current?.some(
+                        condition => condition === attrEntry.attributeCondition.value
+                    ))
+            );
+        });
+        if (isAnythingChanged) {
+            update();
+        }
+    }, [props.attributesList, attributeValuesRef.current, attributeConditionsRef.current]);
+
+    useEffect(() => {
+        const contentWrapperNode = contentRef.current;
+
+        if (contentWrapperNode) {
+            const config: MutationObserverInit = {
+                attributes: true,
+                childList: true,
+                subtree: true,
+                attributeFilter: [props.attribute]
+            };
+            const observer = new MutationObserver(mutationList => {
+                const doUpdate = props.attributesList.some((attrEntries: AttributesListType, index: number) =>
+                    mutationList.some(
+                        mutation =>
+                            !(
+                                mutation.type === "attributes" &&
+                                mutation.attributeName === attrEntries.attribute &&
+                                (mutation.target as Element).getAttribute(mutation.attributeName) ===
+                                    attributeValuesRef.current[index]
+                            )
+                    )
+                );
+
+                if (doUpdate) {
+                    update();
+                }
+            });
+            observer.observe(contentWrapperNode, config);
+
+            return () => {
+                attributeValuesRef.current = [];
+                attributeConditionsRef.current = [];
+                observer.disconnect();
+            };
+        }
+    }, [props.attributesList, attributeValuesRef.current, attributeConditionsRef.current, contentRef.current]);
```

Comment:
props.attribute is  not set  as dependency

Reply:
I was using wrong prop, so this is not needed anymore. because we need props.attributesList